Plugin~/WebRTCPlugin/DummyVideoEncoder.cpp

```diff
@@ -7,94 +7,111 @@ namespace unity
 namespace webrtc
 {
 
-    int32_t DummyVideoEncoder::Encode(
-        const webrtc::VideoFrame& frame,
-        const std::vector<webrtc::VideoFrameType>* frameTypes)
+    DummyVideoEncoder::DummyVideoEncoder(IVideoEncoderObserver* observer)
+    {
+        this->m_setKeyFrame.connect(observer, &IVideoEncoderObserver::SetKeyFrame);
+        this->m_setRates.connect(observer, &IVideoEncoderObserver::SetRates);
+    }
+
+    int32_t DummyVideoEncoder::InitEncode(const webrtc::VideoCodec* codec_settings, int32_t number_of_cores, size_t max_payload_size)
+    {
+        if (codec_settings == nullptr)
+        {
+            return WEBRTC_VIDEO_CODEC_ERR_PARAMETER;
+        }
+        // TODO(kazuki):: this encoder should support codecs other than this.
+        if (codec_settings->codecType != webrtc::kVideoCodecH264)
+        {
+            return WEBRTC_VIDEO_CODEC_ERR_PARAMETER;
+        }
+        if (codec_settings->maxFramerate == 0 || codec_settings->width == 0 || codec_settings->height == 0)
+        {
+            return WEBRTC_VIDEO_CODEC_ERR_PARAMETER;
+        }
+        return WEBRTC_VIDEO_CODEC_OK;
+    }
+
+    int32_t DummyVideoEncoder::RegisterEncodeCompleteCallback(webrtc::EncodedImageCallback* callback)
+    {
+        this->callback = callback;
+        return WEBRTC_VIDEO_CODEC_OK;
+    }
+
+    int32_t DummyVideoEncoder::Release()
```

Comment:
Can we release these too ?
```
        this->m_setKeyFrame.connect(observer, &IVideoEncoderObserver::SetKeyFrame);
        this->m_setRates.connect(observer, &IVideoEncoderObserver::SetRates);
```

Reply:
OK, will fix it

Reply:
Added these lines into `Release` method.
```
this->m_setKeyFrame.disconnect_all();
this->m_setRates.disconnect_all();
```